ui/src/app/views/survey-details.tsx

```diff
@@ -77,7 +84,17 @@ export const SurveyDetails = withCurrentWorkspace()(
         const {workspace, surveyName} = this.props;
         const surveys = await conceptsApi().getSurveyQuestions(
           workspace.namespace, workspace.id, surveyName);
-        this.setState({surveyList: surveys, loading: false});
+        const seeMyAnsList = [];
+        surveys.forEach(survey => seeMyAnsList.push(false));
+        const surveyDetails = [];
+        surveys.map((survey) => {
```

Comment:
`.map` in general should be used for mutating an array. I would probably do this as:
```
const surveyDetails = surveys.map((survey => {
  return {
    question: survey,
    answer: []
  })
```

(Or even cut out the intermediate variable and just put the map in the setState)